File: client-app/src/models/assignedTransaction.ts
/**
 * Similar to a Transaction, but represents a 
 * transaction assigning money to a BudgetItem
 */
export interface IAssignedTransaction {
    id: string;
    budgetItemId: string;
    date: Date;
    amount: number;
}

/**
 * AssignedTransaction class to manage budget item assignments
 */
export default class AssignedTransaction implements IAssignedTransaction {
    id: string;
    budgetItemId: string;
    date: Date;
    amount: number;

    /**
     * Creates a new AssignedTransaction
     * @param id The id of the transaction
     * @param budgetItemId The BudgetItem Id
     * @param date The date money was assigned
     * @param amount The amount assigned
     */
    constructor(id: string, budgetItemId: string, date: Date, amount: number) {
        this.id = id;
        this.budgetItemId = budgetItemId;
        this.date = date;
        this.amount = amount;
    }
}

File: Domain/Models/Budgets/AssignedTransaction.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Models.Transactions;

namespace Domain.Models.Budgets
{
    /// <summary>
    /// Similar to a <see cref="Transaction"/>, but represents a 
    /// transaction assigning money to a <see cref="BudgetItem"/>
    /// </summary>
    public class AssignedTransaction
    {
        /// <summary>
        /// The id
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// The <see cref="BudgetItem"/> Id
        /// </summary>
        public Guid BudgetItemId { get; set; }

        /// <summary>
        /// The date money was assigned
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// the amount assigned
        /// </summary>
        public double Amount { get; set; }

        /// <summary>
        /// The primary account ID
        /// </summary>
        [NotMapped]
        public Guid PrimaryAccountId { get; set; }
    }
}